src/services/fdp-storage/fdp-storage.utils.ts

```diff
@@ -1,19 +1,48 @@
 import { assertBeeUrl } from '../../messaging/message.asserts'
 import { DappId } from '../../model/general.types'
 
-export function dappUrlToId(url: string, beeUrl: string): DappId {
-  assertBeeUrl(beeUrl)
+function constructSubdomainRegex(beeUrl: string): RegExp {
+  const [protocol, domain] = beeUrl.split('://')
 
+  // extracts dApp ENS name from a swarm subdomain link (e.g http://ENS.swarm.localhost:1633...)
+  return new RegExp(
+    `${protocol}://(.+).swarm.${domain.endsWith('/') ? domain.substring(0, domain.length - 1) : domain}.*`,
```

Comment:
why `/` is important to be handled when after `domain` can stay anything?

Reply:
Yes, but the '/' can be optional. For example, if domain is `localhost:1633/`, then the regex should catch URL like `http://ens.swarm.localhost:1633` (without '/').